oraclizeAPI_0.4.sol

```diff
@@ -853,10 +853,10 @@ contract usingOraclize {
 
     modifier oraclize_randomDS_proofVerify(bytes32 _queryId, string _result, bytes _proof) {
         // Step 1: the prefix has to match 'LP\x01' (Ledger Proof version 1)
-        if ((_proof[0] != "L")||(_proof[1] != "P")||(_proof[2] != 1)) throw;
+        require((_proof[0] == "L") && (_proof[1] == "P") && (_proof[2] == 1));
 
         bool proofVerified = oraclize_randomDS_proofVerify__main(_proof, _queryId, bytes(_result), oraclize_getNetworkName());
-        if (proofVerified == false) throw;
+        require(proofVerified);
```

Comment:
If you read here, it is actually the opposite: the require-style exception uses the new revert opcode, while the assert uses the throw opcode. So, it is should be require.

Reply:
yes, I understand, hence why I'm recommending assert. Otherwise, Oraclize can reward itself for providing wrong queries, with proofs that purposefully fail, taking leftover gas as profit.

Reply:
Right, but it could be useful for offchain payments and gas refunds.